solr/solrj-streaming/src/java/org/apache/solr/client/solrj/io/stream/SelectStream.java

```diff
@@ -308,6 +327,12 @@ public Tuple read() throws IOException {
       workingForEvaluators.put(fieldName, original.get(fieldName));
       if (selectedFields.containsKey(fieldName)) {
         workingToReturn.put(selectedFields.get(fieldName), original.get(fieldName));
+      } else {
+        for (String globPattern : selectedFieldGlobPatterns) {
+          if (FilenameUtils.wildcardMatch(fieldName, globPattern)) {
+            workingToReturn.put(fieldName, original.get(fieldName));
+          }
+        }
```

Comment:
So due to the if/else, if the field is explicitly specified, it will not be returned as associated under the glob pattern? That's probably what we want.

The only potential edge case could be around the fact that the if/else is wrt the returned field label. So iiuc if you happened to use an `as [label]` label that matches a field name, that field name would not be returned under any applicable glob spec. This is probably _still_ what we want, but it is a bit weird, and is an edge case that (I think?) was not relevant prior to the addition of glob patterns.

Reply:
I think this is the correct behavior we want, in this case we only want to match provided field names and not labels for those fields.

There are probably generally some edge cases already here if you use `as [label]` where label is also a field name if that field name is separately requested.